mssql/index.d.ts

```diff
@@ -204,22 +220,23 @@ export declare class Request extends events.EventEmitter {
     public constructor(connection?: Connection);
     public constructor(transaction: Transaction);
     public constructor(preparedStatement: PreparedStatement);
-    public execute(procedure: string): Promise<recordSet>;
-    public execute<Entity>(procedure: string, callback: (err?: any, recordsets?: Entity[], returnValue?: any, rowsAffected?: number) => void): void;
-    public input(name: string, value: any): void;
-    public input(name: string, type: any, value: any): void;
-    public output(name: string, type: any, value?: any): void;
-    public pipe(stream: NodeJS.WritableStream): void;
-    public query(command: string): Promise<void>;
-    public query<Entity>(command: string): Promise<Entity[]>;
-    public query(command: string, callback: (err?: any, recordset?: any, rowsAffected?: number) => void): void;
-    public query<Entity>(command: string, callback: (err?: any, recordset?: Entity[]) => void): void;
-    public batch(batch: string): Promise<recordSet>;
-    public batch<Entity>(batch: string): Promise<Entity[]>;
-    public batch(batch: string, callback: (err?: any, recordset?: any) => void): void;
-    public batch<Entity>(batch: string, callback: (err?: any, recordset?: Entity[]) => void): void;
-    public bulk(table: Table): Promise<void>;
-    public bulk(table: Table, callback: (err: any, rowCount: any) => void): void;
+    public execute(procedure: string): Promise<IRecordSet<any>>;
+    public execute<Entity>(procedure: string): Promise<IRecordSet<Entity>>;
```

Comment:
I don't think the type parameter `Entity` here does anything. If only used for return types, it will be ignored because we don't differentiate signatures just by return types